File: tests/exercises/echo-function/evaluation/Evaluator.java
import java.util.*;

public class Evaluator {

    public static EvaluationResult evaluate(Object actual) {
        var correct = "correct".equals(actual);
        return EvaluationResult.builder(correct)
                .withReadableExpected("correct")
                .withReadableActual(actual != null ? actual.toString() : "")
                .withMessage(new EvaluationResult.Message("Hallo"))
                .build();
    }

    public static EvaluationResult evaluateSum(Object actual, Integer sum) {
        var correct = sum == 10;
        return EvaluationResult.builder(correct)
                .withReadableExpected("correct")
                .withReadableActual(actual != null ? actual.toString() : "")
                .withMessage(new EvaluationResult.Message("Hallo"))
                .build();
    }
}

File: tests/exercises/echo-function/evaluation/Evaluator.cs
using Tested;
using System.Collections;

public class Evaluator {
    public static EvaluationResult Evaluate(Object actual) {
        var correct = "correct" == actual;
        var messages = new List<Message>() {new Tested.Message("Hallo")};
        return new EvaluationResult(correct, "correct", actual != null ? actual.ToString() : "", messages);
    }

    public static EvaluationResult EvaluateSum(Object actual, int sum) {
            var correct = sum == 10;
            var messages = new List<Message>() {new Tested.Message("Hallo")};
            return new EvaluationResult(correct, "correct", actual != null ? actual.ToString() : "", messages);
        }
}
